File: 0389-find-the-difference/0389-find-the-difference.php
class Solution {

    /**
     * @param String $s
     * @param String $t
     * @return String
     */
    function findTheDifference($s, $t) {
        $counter = [];
        
        for ($i = 0; $i < strlen($s); $i++) {
            $ch = $s[$i];
            if (array_key_exists($ch, $counter)) {
                $counter[$ch]++;
            } else {
                $counter[$ch] = 1;
            }
        }

        for ($i = 0; $i < strlen($t); $i++) {
            $ch = $t[$i];
            if (!array_key_exists($ch, $counter) || $counter[$ch] == 0) {
                return $ch;
            }
            $counter[$ch]--;
        }
        return '';
    }
}

File: 0389-find-the-difference/0389-find-the-difference.py
from collections import defaultdict

class Solution(object):
    def findTheDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: str
        """
        char_count = defaultdict(int)

        for char in s:
            char_count[char] += 1

        for char in t:
            if char_count[char] == 0:
                return char
            char_count[char] -= 1